lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,327 @@
+                iris.cube.Cube arguments of lightning probability and
+                vertically integrated ice and returns an updated lightning
+                probability cube.
+
+            debug (boolean):
+                True results in verbose output for debugging purposes.
+
+        """
+        self.debug = debug
+        self.radius = radius
+        lead_times = [0., 6.]
+        radii = [self.radius, 2*self.radius]
+        self.neighbourhood = NeighbourhoodProcessing(
+            'circular', radii, lead_times=lead_times)
+
+        # Lightning-rate threshold for Lightning Risk 2 level
+        # Lightning-rate threshold for Lightning Risk 1 level
+        # (dependent on forecast-length)
+        self.lrt_lev1, self.lrt_lev2 = lightning_thresholds
+        # Prob(lightning) value for Lightning Risk 1 & 2 levels
+        self.pl_dict = problightning_values
+
+        if precip_method is None:
+            self.precip_plugin = ApplyPrecip(problightning_values=self.pl_dict)
+        else:
+            self.precip_plugin = precip_method
+        self.ice_plugin = ice_method
+
+    def __repr__(self):
+        """
+        Docstring to describe the repr, which should return a
+        printable representation of the object.
+        """
+        return """<NowcastLightning: radius={radius}, debug={debug},
+ lightning mapping (lightning rate in "min^-1"):
+   upper: lightning rate {lthru} => min lightning prob {lprobu}
+   lower: lightning rate {lthrl} => min lightning prob {lprobl}
+With:
+{precip}
+{ice}
+>""".format(radius=self.radius, debug=self.debug,
+            lthru=self.lrt_lev1, lthrl=self.lrt_lev2,
+            lprobu=self.pl_dict[1], lprobl=self.pl_dict[2],
+            precip=self.precip_plugin,
+            ice=self.ice_plugin)
+
+    def _process_haloes(self, cube):
+        """
+        Adjust data so that lightning probability does not decrease too rapidly
+        with distance.
+
+        Args:
+            cube (iris.cube.Cube):
+                Radius will be applied equally on x and y dimensions.
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube of same shape as cube with haloes applied.
+        """
+        new_cube = self.neighbourhood.process(cube.copy())
+        return new_cube
+
+    def _update_meta(self, cube):
+        """
+        Modify the meta data of input cube to resemble a Nowcast of lightning
+        probability
+
+        Args:
+            cube (iris.cube.Cube):
+                An input cube
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube - a copy of input cube with meta-data relating to
+                a Nowcast of lightning probability.
+                The data array will be a copy of the input cube.data
+        """
+        new_cube = cube.copy()
+        new_cube.rename("lightning_probability")
+        new_cube.remove_coord('threshold')
```

Comment:
What happens if the input cube has no 'threshold' coordinate?  Do we need a note in the docstring?

Reply:
I will add a unit-test to make sure that a useful error is generated.

Reply:
Sorry: I think I meant it would be useful to say in the docstring that the input cube is expected to have a "threshold" coordinate, and / or to have a `Raises` block in the docstring.